resources/osl_matrix.rb

```diff
@@ -0,0 +1,92 @@
+resource_name :osl_matrix
+provides :osl_matrix
+unified_mode true
+
+default_action :create
+
+property :app_services, Array, default: []
+property :config, String
+property :domain, String, name_property: true
+property :pg_host, String
+property :pg_name, String
+property :pg_username, String
+property :pg_password, String
+property :tag, String, default: 'latest'
+property :use_sqlite, [true, false], default: false
+
+action :create do
+  require 'securerandom'
+
+  # Initalize the synapse user
+  user 'synapse-host' do
+    system true
+  end
+
+  # Synapse configuration
+  directory '/srv/synapse' do
+    owner 'synapse-host'
+    mode '750'
+  end
+
+  # Keys directory
+  directory '/srv/synapse/keys' do
+    owner 'synapse-host'
+    mode '700'
+  end
+
+  # Create the docker network for Synapse-related items
+  docker_network 'synapse-network'
+
+  # Generate secret keys for Synapse. Only generate once.
+  file '/srv/synapse/keys/registration.key' do
+    content SecureRandom.base64(32)
```

Comment:
Please convert this to a simple helper method and remove the require above which will get added to the helper instead.